mm/2s2h/Enhancements/Trackers/ItemTracker/ItemTracker.cpp

```diff
@@ -0,0 +1,469 @@
+            imageObject.textureId = Ship::Context::GetInstance()->GetWindow()->GetGui()->GetTextureByName(
+                (const char*)gItemIcons[gSaveContext.save.saveInfo.inventory.items[bottleId]]);
+        } else {
+            imageObject.textureId = Ship::Context::GetInstance()->GetWindow()->GetGui()->GetTextureByName(
+                (const char*)gItemIcons[ITEM_BOTTLE]);
+        }
+        imageObject.textureColor.w = gSaveContext.save.saveInfo.inventory.items[bottleId] != ITEM_NONE ? 1 : 0.4f;
+    } else if (itemId == ITEM_SKULL_TOKEN_SWAMP || itemId == ITEM_SKULL_TOKEN_OCEAN) {
+        uint32_t tokenCount = 0;
+        if (itemId == ITEM_SKULL_TOKEN_SWAMP) {
+            tokenCount = Inventory_GetSkullTokenCount(SCENE_KINSTA1);
+        } else {
+            tokenCount = Inventory_GetSkullTokenCount(SCENE_KINDAN2);
+        }
+        imageObject.textureColor.w = tokenCount > 0 ? 1 : 0.4f;
+    } else if (itemId >= ITEM_WOODFALL_STRAY_FAIRY && itemId <= ITEM_STONE_TOWER_STRAY_FAIRY) {
+        imageObject.textureId = Ship::Context::GetInstance()->GetWindow()->GetGui()->GetTextureByName(
+            (const char*)fairyIconTextures[itemId - ITEM_WOODFALL_STRAY_FAIRY]);
+        imageObject.textureColor.w =
+            gSaveContext.save.saveInfo.inventory.strayFairies[itemId - ITEM_WOODFALL_STRAY_FAIRY] > 0 ? 1 : 0.4f;
+    } else if (itemId >= ITEM_WOODFALL_DUNGEON_MAP && itemId <= ITEM_STONE_TOWER_KEY_BOSS) {
+        const int dungeonIndex = (itemId - ITEM_WOODFALL_DUNGEON_MAP) / 4;
+        const int itemTypeIndex = (itemId - ITEM_WOODFALL_DUNGEON_MAP) % 4;
+        switch (itemTypeIndex) {
+            case 0:
+                imageObject.textureColor.w = CHECK_DUNGEON_ITEM(DUNGEON_COMPASS, dungeonIndex) ? 1.0f : 0.4f;
+                break;
+            case 1:
+                imageObject.textureColor.w = CHECK_DUNGEON_ITEM(DUNGEON_MAP, dungeonIndex) ? 1.0f : 0.4f;
+                break;
+            case 2:
+                imageObject.textureColor.w = DUNGEON_KEY_COUNT(dungeonIndex) > 0 ? 1.0f : 0.4f;
+                break;
+            case 3:
+                imageObject.textureColor.w = CHECK_DUNGEON_ITEM(DUNGEON_BOSS_KEY, dungeonIndex) ? 1.0f : 0.4f;
+                break;
+        }
+    } else {
+        imageObject.textureColor.w = INV_CONTENT(itemId) != ITEM_NONE ? 1 : 0.4f;
+    }
+
+    return imageObject;
+}
+
+std::string GetItemCounts(int16_t itemId, bool isRandoItem) {
+    std::string countStr = "";
+    if (isRandoItem) {
+        if (itemId == RI_TRIFORCE_PIECE) {
+            int16_t maxPieces = gSaveContext.save.shipSaveInfo.rando.randoSaveOptions[RO_TRIFORCE_PIECES_REQUIRED];
+            countStr = std::to_string(gSaveContext.save.shipSaveInfo.rando.foundTriforcePieces).c_str();
+            countStr += "/";
+            if (maxPieces == 1000) {
+                countStr += "1k";
+            } else {
+                countStr = std::to_string(maxPieces).c_str();
```

Comment:
If there aren't 1000 Triforce pieces, you won't actually see your current count.
```suggestion
                countStr += std::to_string(maxPieces).c_str();
```